suitcase/utils/conftest.py

```diff
@@ -0,0 +1,123 @@
+import bluesky # noqa
+from bluesky.tests.conftest import RE # noqa
+from bluesky.plans import count # noqa
+from bluesky.plan_stubs import trigger_and_read, configure
+import event_model # noqa
+from ophyd.tests.conftest import hw # noqa
+import pytest # noqa
+from . import UnknownEventType # noqa
+
+
+# Some useful plans for use in testing
+def simple_plan(dets):
+    '''A simple plane which runs count with num=5'''
+    yield from count(dets, num=5)
+
+
+def multi_stream_one_descriptor_plan(dets):
+    '''A plan that has two streams but on descriptor per stream)'''
+    @bluesky.preprocessors.baseline_decorator(dets)
+    def _plan(dets):
+        yield from count(dets)
+
+    yield from _plan(dets)
+
+
+def one_stream_multi_descriptors_plan(dets):
+    '''A plan that has one stream but two descriptors per stream)'''
+    @bluesky.preprocessors.run_decorator()
+    def _internal_plan(dets):
+        yield from trigger_and_read(dets)
+        for det in dets:
+            yield from configure(det, {})
+        yield from trigger_and_read(dets)
+
+    yield from _internal_plan(dets)
+
+
+@pytest.fixture(params=[['det'], ['direct_img'], ['det', 'direct_img']],
+                scope='function')
+def detector_list(hw, request):
+    dets = [getattr(hw, det_name) for det_name in request.param]
+    return dets
+
+
+@pytest.fixture(params=['event', 'bulk_events', 'event_page'],
+                scope='function')
+def event_type(request):
+    return  request.param
+
+
+@pytest.fixture()
+def events_data(RE, detector_list, event_type):
```

Comment:
Can we also parameter this by plan so the consumer plans just have to ask for `event_data` and automatically get the plan multiplexing?  An advantage of this is that if we add a new plan here, we will immediantly get it tested in all of the suitcases (which can then, if they can't handle it, explicitly xfail that plan), but we don't have to remember to manually go add the tests to each and every suitcase. 

This is reminiscent of how we currently run the databroker tests (have a standard set of tests, each sub-component provides it's own version of the fixtures), but I think this is a better way for suitcase (have a standard fixture, each suitcase has custom tests).

Either that or can we rename this to `generate_data`?  This is currently named an a noun, but we use it as a verb.  These things seem trivial, but for code that is used by lots of people or last a long time (and hopefully this will be both!) these subtle hints really do help.

Reply:
@tacaswell, @danielballan had a different opinion here as it may be that we do not want to run every suitcase against every plan (they are more like a pick which ones are relevant approach). I am happy to implement either approach.

Reply:
renaming and reparametrizing is done